pallets/swaps/src/tests.rs

```diff
@@ -490,3 +490,87 @@ mod fulfill {
 		});
 	}
 }
+
+mod zero_amount_order {
+	use super::*;
+
+	#[test]
+	fn when_apply_over_no_swap() {
+		new_test_ext().execute_with(|| {
+			MockTokenSwaps::mock_place_order(|_, _, _, _, _| {
+				panic!("this mock should not be called");
+			});
+
+			assert_ok!(
+				<Swaps as TSwaps<AccountId>>::apply_swap(
+					&USER,
+					SWAP_ID,
+					Swap {
+						currency_in: CURRENCY_B,
+						currency_out: CURRENCY_A,
+						amount_out: 0,
+					},
+				),
+				SwapStatus {
+					swapped: 0,
+					pending: 0,
+				}
+			);
+
+			assert_eq!(OrderIdToSwapId::<Runtime>::get(ORDER_ID), None);
+			assert_eq!(SwapIdToOrderId::<Runtime>::get((USER, SWAP_ID)), None);
+		});
+	}
+
+	#[test]
+	fn when_apply_over_smaller_inverse_swap_but_math_precission() {
```

Comment:
We do not really need this test having `pallet-swaps` tests. But I wanted to force a use case from FI PoV. We can finally remove it once we're sure it's fully solved.